browser/views.py

```diff
@@ -220,24 +221,43 @@ def _handle_known_person_form(request, extra_form, person):
 
 
 @staff_member_required
-def edit_person(request, person_id):
-    person = get_object_or_404(Person, pk=person_id)
+def edit_person(request, person_id=None):
+    person = None
     if request.method == 'GET':
-        form = PersonForm(instance=person)
-        extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        if person_id:
+            person = get_object_or_404(Person, pk=person_id)
+            form = PersonForm(instance=person)
+            extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        else:
+            form = PersonForm()
+            extra_form = KnownPersonForm()
+
+    if request.method == 'POST':
+        if person_id:
+            person = get_object_or_404(Person, pk=person_id)
+            form = PersonForm(request.POST, instance=person)
+            extra_form = KnownPersonForm(request.POST, instance=person.authority, prefix='known')
+            if form.is_valid():
+                person = form.save()
+        else:
+            form = PersonForm(request.POST)
+            extra_form = KnownPersonForm(request.POST)
+            if form.is_valid():
+                person = Person.objects.create(
+                    changed_by=request.user,
+                    first_name=form.cleaned_data.get('first_name'),
+                    last_name=form.cleaned_data.get('last_name')
+                )
+
+        if person.validated and person.validated_by is None:
+            person.validated_by = request.user
+            person.validated_on = datetime.datetime.now()
+            person.save()
+
+        if extra_form.is_valid():
+            _handle_known_person_form(request, extra_form, person)
+        return HttpResponseRedirect(reverse('person', args=(person.id,)))
```

Comment:
what if the form is not valid, should it go back to the edit screen?

Reply:
As per the existing code even if the <b>extra_form</b> is not valid, the user is redirected to the person page. I think that is why the above code is correct.

Reply:
Currently, fields are validated in the HTML. You still have to make sure this code works correctly though (redirecting back to the form if form is not valid), to make sure it doesn't break in the future (e.g. we're adding another form somewhere else using the same code).

Reply:
I have added the condition to redirect to edit/create person screen if the form is invalid.